compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/arguments/K2JVMCompilerArguments.kt

```diff
@@ -511,6 +511,13 @@ Also sets `-jvm-target` value equal to the selected JDK version"""
     )
     var linkViaSignatures: Boolean by FreezableVar(false)
 
+    @Argument(
+        value = "-Xenable-debug-mode",
```

Comment:
`-Xdebug`?

Reply:
I'd recommend `-Xdebug-mode`.

`-Xdebug` might give an impression that we're going to debug the compiler.

Reply:
Disagree. `--debug` is already used in compilers:
1. https://developer.arm.com/documentation/dui0493/g/linker-command-line-options/--debug----no-debug 
1. https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html
1. https://docs.microsoft.com/en-us/cpp/build/reference/debug-generate-debug-info?view=msvc-170

Swift uses `-DDEBUG`:
1. https://kitefaster.com/2016/01/23/how-to-specify-debug-and-release-flags-in-xcode-with-swift/

Gradle uses `debug`:
1. https://docs.gradle.org/current/dsl/org.gradle.api.tasks.compile.CompileOptions.html#org.gradle.api.tasks.compile.CompileOptions:debug

It is less common, than `-g`, but still, a simple `--debug` flag is a common compiler flag.

While `--debug-mode` is used in one test suite (I cannot find a usage for compilers)
1. https://testcafe.io/documentation/402835/guides/basic-guides/debug

Reply:
> To test, you should check fields of the continuation object (L$ fields are enough)


It would be nice to implement these tests on the debugger side. Should we also add some compiler tests that verify generated bytecode with that key enabled like `org.jetbrains.kotlin.test.runners.codegen.BytecodeTextTestGenerated.Coroutines.Cleanup`?

Reply:
@ilmirus OK, I agree. I guess I was thinking of Maven, which used `-Xdebug` exactly in this sense, but looks like it's not intended to be used anymore.

Reply:
@nikita-nazarov There should still be backend tests, yes. I should rewrite clean-up tests, though, since I do not like bytecode text tests, which are fragile. Instead I will use java reflection to check `L$` fields of the continuation.